daemonset.yaml

```diff
@@ -60,40 +60,81 @@ data:
     #!/bin/bash
 
     OS_TYPE=`uname -a`
-    RPM_TYPE=""
-    echo "Detecting OS..."
+    VERSION_AMZN="1.1-12"
+    PKG_AL1="log4j-cve-2021-44228-hotpatch-$VERSION_AMZN.amzn1.noarch"
+    PKG_AL2="log4j-cve-2021-44228-hotpatch-$VERSION_AMZN.amzn2.noarch"
+    VERSION_DEBIAN="1.1.12" #Change for debian versions
+    PKG_DEBIAN="log4j-cve-2021-44228-hotpatch"
+    ARCH_DEBIAN="all"
+    STATUS_OK_DEBIAN="install ok installed"
+
+    check_file() {
+        echo "Detecting patch file: $1 ..."
+        if test -f "$1"; then
+            echo "$1 exists."
+        else
+            echo "ERROR: $1 does not exist."
+            exit 1
+        fi
+    }
+
+    install_rpm() {
+        echo "Installing $1 ..."
+        eval "rpm -Uvh $1"
+    }
+
+    install_dpkg() {
+        echo "Installing $1 ..."
+        eval "dpkg -i $1"
+    }
+
+    verify_rpm() {
+        echo "Verifying $1 package ..."
+        local OUTPUT=`eval "rpm -V $1"`
+        if [[ -z $OUTPUT ]]; then
+            echo "$1 installed and verified"
+        else
+            echo "$1 could not be verified"
+            echo "$OUTPUT"
+            exit 1
+        fi
+    }
+
+    #Must verify package and version
+    verify_dpkg() {
+        echo "Verifying $1 package, version $2 ..."
+        local OUTPUT=`eval "dpkg -s $1"`
+        if [[ $OUTPUT =~ "Status: $STATUS_OK_DEBIAN" ]] && [[ $OUTPUT =~ "Version: $2" ]]; then
+            echo "$1, version $2, installed and verified"
+        else
+            echo "$1, version $2,  could not be verified"
+            echo "$OUTPUT"
+            exit 1
+        fi
+    }
+
+    echo "Detecting OS ..."
     if [[ "$OS_TYPE" =~ "amzn1" ]]; then
-        RPM_TYPE="amzn1"
-        echo "OS Matched $RPM_TYPE"
+        echo "OS Matched amzn1 - $OS_TYPE"
+        check_file "/tmp/install/$PKG_AL1.rpm"
+        install_rpm "/tmp/install/$PKG_AL1.rpm"
+        verify_rpm "$PKG_AL1"
     elif [[ "$OS_TYPE" =~ "amzn2" ]]; then
-        RPM_TYPE="amzn2"
-        echo "OS Matched $RPM_TYPE"
+        PKG_TYPE="amzn2"
+        echo "OS Matched amzn2 - $OS_TYPE"
+        SUFFIX=".rpm"
```

Comment:
Do we need `SUFFIX=".rpm"`? Let's clean this up before merging.

Reply:
Removed